Report_VirtualBox.md

```diff
@@ -0,0 +1,168 @@
+   
+# 8. Route configuration 
+  ## 1. check the connection between server and gateway using tcp over enp0s8
+    
+  ![tcp A to B](https://user-images.githubusercontent.com/25878224/226161389-0d02c3fb-1718-4d2d-b2bc-6835b16dcaee.PNG)  
+    
+  ## 2. check the connection between client and gateway using tcp over enp0s9
+
+  ![tcp Cto B](https://user-images.githubusercontent.com/25878224/226161414-13ed7a53-02d2-43f1-9083-21312473c52e.PNG)
+
+  ## 3. connection between sever and client over gateway 
+   ### The easiest way with the current configuration is to allow ip packet forwarding in our gateway
+   
+   ![10- ip forward_1 -1 B](https://user-images.githubusercontent.com/25878224/226161470-4e8afa29-c675-41ce-8773-efe240e60747.PNG)
+
+  ![11- ip forward_1 -2 B](https://user-images.githubusercontent.com/25878224/226161472-662aa3c6-bea6-4cbb-beec-3343e2db1502.PNG)
+
+  
+   ### Use the tcpdump command to look at how packets are forwarded on the interfaces enp0s8 (with server) and enp0s9 (with client)
+        
+  ![tcp 8 A to C](https://user-images.githubusercontent.com/25878224/226161725-918f6620-9cf7-4673-a81b-6ac17cec9290.PNG)
+
+  ![tcp9 C to A](https://user-images.githubusercontent.com/25878224/226161728-fde66fe7-94f2-4746-bf09-2b3993debdec.PNG)
+
+  ## 4. To allow only specific packets to be forwarded on a specific port, we configured the route:
+   
+  ![iptables](https://user-images.githubusercontent.com/25878224/226161741-7ec045d6-3ef1-42d8-9ab7-4de01d6ba9c2.PNG)
+
+  ## 5. save the iptables rules
+       
+  ![iptables save rules 0](https://user-images.githubusercontent.com/25878224/226161753-418e00f8-1264-445c-a84b-1474fb45d759.PNG)
+  
+  ![iptables save rules 1](https://user-images.githubusercontent.com/25878224/226161763-e11204a4-27c4-4b3e-a9b7-870d538ebf99.PNG)
+  
+  ![iptables save rules 2](https://user-images.githubusercontent.com/25878224/226161765-6c929389-35f3-47a4-9ac2-f342e7360784.PNG)
+  
+  ![iptables save rules 3](https://user-images.githubusercontent.com/25878224/226161767-2c886a15-f1d6-4c86-8a2f-7780ac98dac6.PNG)
+  
+  ![iptables save rules 4](https://user-images.githubusercontent.com/25878224/226161768-05bb7a03-9861-4660-8020-ad04c3289011.PNG)
+
+
+# 9. Web Server (Machine A) and the requests from client (Machine C) using curl 
+  ## - on the server we create app.py file in new directory called server, then we will raise the server using Flask framework 
+  
+ ![9- app file](https://user-images.githubusercontent.com/25878224/226161789-e800108c-b581-4920-b790-25783b3fd6aa.PNG)
+
+
+  ## - Let's try to send a request from the client to the server and see what is displayed in tcpdump
+ 
+  ![webserver](https://user-images.githubusercontent.com/25878224/226161799-9a06f24e-8e89-49b1-ae3c-99bc3baba0fd.PNG)
+
+  
+  ## - check with wrong port
+   ### if we change the port in web client request, we obtain (Failed connection)
```

Comment:
in this case you should run web server in port 5001 (not 5000), than curl server from client and if firewall is setted well, you request will be reduce